modules/monitor/hardware/resource_monitor.cc

```diff
@@ -37,6 +39,77 @@ DEFINE_double(resource_monitor_interval, 5,
 namespace apollo {
 namespace monitor {
 
+namespace {
+
+bool GetPIDByCmdLine(const std::string& process_dag_path, int* pid) {
+  const std::string system_proc_path = "/proc";
+  const std::string proc_cmdline_path = "/cmdline";
+  const auto dirs = cyber::common::ListSubPaths(system_proc_path);
+  std::string cmd_line;
+  for (const auto& dir_name : dirs) {
+    if (!std::all_of(dir_name.begin(), dir_name.end(), isdigit)) {
+      continue;
+    }
+    *pid = std::stoi(dir_name);
+    std::ifstream cmdline_file(
+        absl::StrCat(system_proc_path, "/", dir_name, proc_cmdline_path));
+    std::getline(cmdline_file, cmd_line);
+    if (cmd_line.find(process_dag_path) != std::string::npos) {
```

Comment:
Try https://abseil.io/docs/cpp/guides/strings#abslstrcontains-for-string-matching